net/speedtest/speedtest.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package speedtest contains both server and client code for
+// running speedtests between tailscale nodes.
+package speedtest
+
+import (
+	"fmt"
+	"math"
+	"time"
+)
+
+const (
+	LenBufData        = 32000 // Size of the block of randomly generated data to send.
+	DefaultNumSeconds = 5     // Default time for a test
+	MaxNumSeconds     = 30
+	Version           = 1
+	increment         = 1.0 // increment to display results for, in seconds
+)
+
+// This is the initial message sent to the sever, that contains information on how to
+// conduct the test.
+type TestConfig struct {
+	Version      int           `json:"version"`
+	TestDuration time.Duration `json:"time"`
+}
+
+// This is the response to the TestConfig message. It contains errors that the server might
+// have with the TestConfig.
+type TestConfigResponse struct {
+	Errors string `json:"errors,omitempty"`
+}
+
+// This represents the Result of a speedtest within a specific interval
+type Result struct {
+	startTime  float64
```

Comment:
startTime is meant to be the start of the interval, in seconds since the start of the test, so would it be okay to make it a time.Duration, or rename it to reflect that? I'm not sure how I'd keep it as a time.time without also including the test's start time for context.

Reply:
Nevermind, I know how I'd do it